### DISCUSSION TITLE
Any best practices for upgrading MongoDB ?

### DISCUSSION THREAD

**Discussion by elpibedeoro** [Ylianst/MeshCentral#6135] [Q&A]
Hi,

How do you manage the upgrading of MongoDB for Meshcentral ?

Thanks.

**Comment by si458** [comment]
Stop meshcentral, upgrade mongodb, restart meshcentral?
Or that's the way I would do it. Then ur data is safe, and doesn't corrupt ur data while it restarts and meshcentral needs to write

**Comment by elpibedeoro** [comment]
Thanks, this is what I did from v. 4.4 to 5.0 and it worked.
But I had a problem while upgrading from version 5.0 to 6.0 and I ran into error 62: needDowngrade this exe can't use the existing data files (https://github.com/mongodb/mongo/blob/master/src/mongo/util/exit_code.h).

I think I might get confused about mongodb that comes with Meshcentral... when I upgrade meshcentral, it comes with mongodb module v. 4.17,  so which version will be used ?

**Reply by si458** [reply]
Erm not a clue?
If u upgraded from 4 to 5 and it still worked then that's good!
Found this, no idea if it might help u or not? https://www.mongodb.com/community/forums/t/any-way-to-recovery-from-error-62-without-losing-data/243582

**Reply by elpibedeoro** [reply]
OK I got it, I made the same mistake in fact ! Now it's ok, I successfully upgrade to v.6 thanks for your precious help.
If it can help someone else, this is how I proceeded for v5 to v6 : 

```
systemctl stop meshcentral
mongosh
	|--> db.adminCommand( { getParameter: 1, featureCompatibilityVersion: 1 } )
	|-->    Ensure that the result includes "featureCompatibilityVersion" : { "version" : "5.0" }
   		If it doesn’t, run the command db.adminCommand( { setFeatureCompatibilityVersion: "5.0" } )
systemctl stop mongod
curl -fsSL https://www.mongodb.org/static/pgp/server-6.0.asc | gpg -o /usr/share/keyrings/mongodb-server-6.0.gpg --dearmor
echo "deb [ signed-by=/usr/share/keyrings/mongodb-server-6.0.gpg ] http://repo.mongodb.org/apt/debian bullseye/mongodb-org/5.0 main" | sudo tee /etc/apt/sources.list.d/mongodb-org.list
apt update && apt upgrade
systemctl start mongod meshcentral
```
What I still don't understand is that Meshcentral install  'mongodb@4.13.0' when upgrading, so I'm a little bit lost. Any idea ?
Thank you !

**Reply by si458** [reply]
That's just the npm module version for nodejs to talk to mongodb! Yes we do need to look at upgrading to package version 6 or above but the is a few breaking changes so its more testing first!
